mlir/include/mlir/Dialect/EmitC/IR/EmitC.td

```diff
@@ -1137,7 +1153,9 @@ def EmitC_AssignOp : EmitC_Op<"assign", []> {
     ```
   }];
 
-  let arguments = (ins EmitCType:$var, EmitCType:$value);
+  let arguments = (ins 
+      Res<EmitC_LValueType, "", [MemWrite<DefaultResource, 1, FullEffect>]>:$var,
+      Res<EmitCType, "", [MemRead<DefaultResource, 0, FullEffect>]>:$value);
```

Comment:
I think `MemRead` doesn't apply to `value` because that one is already SSA and not memory

Reply:
Done

Reply:
> I think `MemRead` doesn't apply to `value` because that one is already SSA and not memory

+1